config/opensearch_dashboards.yml

```diff
@@ -36,6 +36,10 @@
 # Set the value of this setting to true to enable plugin application config. By default it is disabled.
 # application_config.enabled: false
 
+# Set the value of this setting to true to enable plugin CSP handler. By default it is disabled.
+# It requires the application config plugin as its dependency.
+# csp_handler.enabled: false
```

Comment:
Why need extra configuration? Can we enable by default and if user adds record if config index and this plugin should enforce csp rules otherwise do nothing?

Reply:
Once the feature is enabled which means enforcing Clickjacking mitigation, it will become a breaking change. We have received guidance from product team to keep the feature disabled in 2.x and only enable in 3.x https://github.com/opensearch-project/OpenSearch-Dashboards/issues/5639#issuecomment-1930756997

So, to your question, the extra configuration is for the switch on/off.

Reply:
We will enable by default for sure in 3.x. This is to ensure that the product the org ships is secure by default. Whether we remove it could be decided later.